integration-tests/events/lib.rs

```diff
@@ -0,0 +1,189 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract]
+pub mod events {
+    #[ink(storage)]
+    pub struct Events {
+        value: bool,
+    }
+
+    impl Events {
+        /// Creates a new events smart contract initialized with the given value.
+        #[ink(constructor)]
+        pub fn new(init_value: bool) -> Self {
+            Self { value: init_value }
+        }
+
+        /// Flips the current value of the boolean.
+        #[ink(message)]
+        pub fn flip(&mut self) {
+            self.value = !self.value;
+            self.env()
+                .emit_event(event_def::Flipped { value: self.value })
+        }
+
+        /// Emit an event with a 32 byte topic.
+        #[ink(message)]
+        pub fn emit_32_byte_topic_event(&mut self, maybe_hash: Option<[u8; 32]>) {
+            self.env().emit_event(event_def::ThirtyTwoByteTopics {
```

Comment:
My guess would be that for the new crate you failed to add your new crate to the `std` feature of the contract. Otherwise the `EventMetadata` won't be derived.

https://github.com/paritytech/ink/blob/a78360ad1a3bea244fb686592836eb04fea3470f/integration-tests/events/Cargo.toml#L27

Perhaps we could add a check to the `cargo-contract` companion https://github.com/paritytech/cargo-contract/pull/1184 which would prevent this mistake. It would check that all dependencies are added to the `std` feature.

Reply:
It has not solved an issue for ,e

Reply:
Maybe it is something to do with MacOS. I have added some tests to `integration_tests/events` to ensure that all events that are defined and used are included in the metadata. Can you try and run them on your machine?

Reply:
Nope. Still has not resolved the issue. I did `cargo clean` then `cargo contract build` and here is the dump of events:
```json
    "events": [
      {
        "args": [
          {
            "docs": [],
            "indexed": false,
            "label": "value",
            "type": {
              "displayName": [],
              "type": 0
            }
          }
        ],
        "docs": [],
        "label": "InlineFlipped",
        "module_path": "events::events",
        "signature_topic": "0xab768ef77e8fd0c0740c523252aebd65d37bb2b5f508d2306f876396ca7e4447"
      },
      {
        "args": [
          {
            "docs": [],
            "indexed": false,
            "label": "value",
            "type": {
              "displayName": [],
              "type": 0
            }
          }
        ],
        "docs": [],
        "label": "ForeignFlipped",
        "module_path": "event_def",
        "signature_topic": "0x7ce035d0d27075652696f71842317eb893587fc339f72377127bb4cfd293cf03"
      },
      {
        "args": [
          {
            "docs": [],
            "indexed": true,
            "label": "hash",
            "type": {
              "displayName": [],
              "type": 6
            }
          },
          {
            "docs": [],
            "indexed": true,
            "label": "maybe_hash",
            "type": {
              "displayName": [],
              "type": 5
            }
          }
        ],
        "docs": [],
        "label": "ThirtyTwoByteTopics",
        "module_path": "event_def",
        "signature_topic": "0xa48606fe432343ce5000004df20ae1efc75fc1dfd9ecb9b99006df6b671faf00"
      }
    ],
 ```

Reply:
Please try with https://github.com/paritytech/cargo-contract/pull/1200, it should fix that.